src/lib.rs

```diff
@@ -11,10 +11,10 @@ pub use sqlx_core::describe::Describe;
 pub use sqlx_core::executor::{Execute, Executor};
 pub use sqlx_core::from_row::FromRow;
 pub use sqlx_core::pool::{self, Pool};
-pub use sqlx_core::query::{query, query_with};
+pub use sqlx_core::query::{query, query_with, query_with_result};
 pub use sqlx_core::query_as::{query_as, query_as_with};
 pub use sqlx_core::query_builder::{self, QueryBuilder};
-pub use sqlx_core::query_scalar::{query_scalar, query_scalar_with};
+pub use sqlx_core::query_scalar::{query_scalar, query_scalar_with, query_scalar_with_result};
```

Comment:
I would make these separate `#[doc(hidden)]` reexports so it's clear from reading the source that they're not part of the public API.